include/envoy/server/listener_manager.h

```diff
@@ -102,6 +102,20 @@ class ListenerComponentFactory {
  */
 class ListenerManager {
 public:
+  // Criteria for selecting listeners to stop.
+  struct StopListenerSelector {
+
+    enum class ListenerDirection {
```

Comment:
nit: I don't think you need the outer struct. Can you just call this enum `StopListenersType` or something and use that as the param to the `stopListeners` function?

Reply:
As I mentioned the outer struct was added to support for other criteria in future like name - I am OK removing that for now and adding it later when it is needed, if you think that is best

Reply:
Yeah let's remove for now.